assets/js/custom/Translation.js

```diff
@@ -2,6 +2,111 @@
 
 // eslint-disable-next-line no-unused-vars
 function Translation (hasDescription, hasCredit) {
+  let languageMap = {}
+  let targetLanguage = document.documentElement.lang
+
+  $(document).ready(function () {
+    $.ajax({
+      url: '../languages',
+      type: 'get',
+      success: function (data) {
+        languageMap = data
+      }
+    })
+  })
+
+  $('.mdc-list-item__text').each(function () {
+    if ($(this).parent()[0].hasAttribute('selected')) {
+      targetLanguage = fixLanguageCode($(this).parent().attr('data-value'))
+    }
+  })
+
+  $(document).on('click', '.comment-translation-button', function () {
+    const commentId = $(this).attr('id').substring('comment-translation-button-'.length)
+
+    $(this).hide()
+
+    if ($('#comment-text-translation-' + commentId).attr('lang') !== targetLanguage) {
+      $('#comment-translation-loading-spinner-' + commentId).show()
+      translateComment(commentId)
+    } else {
+      openTranslatedComment(commentId)
+    }
+  })
+
+  $(document).on('click', '.remove-comment-translation-button', function () {
+    const commentId = $(this).attr('id').substring('remove-comment-translation-button-'.length)
+    $(this).hide()
+    $('#comment-translation-button-' + commentId).show()
+    $('#comment-translation-wrapper-' + commentId).slideUp()
+    $('#comment-text-wrapper-' + commentId).slideDown()
+  })
+
+  function fixLanguageCode (languageCode) {

```

Comment:
When do you need to fix the language code?

Reply:
I am getting the language code from the language selector in the footer which lists the language code as en_BR instead of en-BR so it needs to be changed sometimes. Also, if I get it from the document.lang property it only shows language codes of size 2 which is why I am getting it from the selector.

Reply:
have you tried this? https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/language
i think getting the language from the footer is not good practice. if they change how the footer works, it'd be hard to remember also to update translation stuff

Reply:
I've done some testing with the navigator and even if you select a different language in the footer it sometimes returns the wrong language. Just from some quick testing, it looks like it detects all the languages on the page and then returns the most frequently used language which sometimes isn't the user's selected language in the footer.

Reply:
Added logic to get target language from cookie